test/e2e/scheduling/priorities.go

```diff
@@ -304,18 +306,18 @@ var _ = SIGDescribe("SchedulerPriorities [Serial]", func() {
 		nodeName := nodeList.Items[0].Name
 
 		ginkgo.By("Trying to apply 10 (tolerable) taints on the first node.")
+		// Defer cleaning up the taints to guarantee that all matching taints get removed after the test
+		defer cleanupTestTaints(cs, nodeName)
```

Comment:
to avoid replicating the remove taint code, we could create the taints here in a list

```
tolerableTaints := getRandomTaints(len(nodelList.Items))
intolerableTaints := getRandomTaints(len(nodelList.Items))
defer e2enode.RemoveTaintOffNode(cs, node.Name, tolerableTaints...)
defer e2enode.RemoveTaintOffNode(cs, node.Name, intolerableTaints...)
```

and then below just use those taints to apply to the nodes.

Reply:
I'm not sure what you mean by your code example, but we are only trying to add taints to 1 node (so `len(nodeList.Items)` shouldn't be necessary)

Do you want to refactor `addRandomTaintToNode` to just return the taints, instead of actually adding them? Or introduce a new function `getRandomTaints` that does that?

Also, `e2enode.RemoveTaintOffNode` only accepts 1 taint, so we could write a wrapper for it that takes a list.

Reply:
> I'm not sure what you mean by your code example, but we are only trying to add taints to 1 node (so len(nodeList.Items) shouldn't be necessary)

I see line 320 adds taints to the rest of the of the nodes.

> Do you want to refactor addRandomTaintToNode to just return the taints, instead of actually adding them? Or introduce a new function getRandomTaints that does that?

`getRandomTaint` would basically be:
```
func getRandomTaint() {
 return v1.Taint{
		Key:    fmt.Sprintf("kubernetes.io/scheduling-priorities-e2e-taint-key-%s", string(uuid.NewUUID())),
		Value:  fmt.Sprintf("testing-taint-value-%s", string(uuid.NewUUID())),
		Effect: v1.TaintEffectPreferNoSchedule,
	}
}
```

addRandomTaintToNode becomes addTaintToNode:
```
func addRandomTaintToNode(cs clientset.Interface, nodeName string, taint* v1.Taint) {
	e2enode.AddOrUpdateTaintOnNode(cs, nodeName, *taint)
	framework.ExpectNodeHasTaint(cs, nodeName, taint)
}
```

> Also, e2enode.RemoveTaintOffNode only accepts 1 taint, so we could write a wrapper for it that takes a list.

ah, I was looking at the function under [staging/src/k8s.io/cloud-provider/node/helpers/taints.go](https://github.com/kubernetes/kubernetes/blob/cea1d4e20b4a7886d8ff65f34c6d4f95efcb4742/staging/src/k8s.io/cloud-provider/node/helpers/taints.go#L147), yeah we could wrap it.

Reply:
Added the wrapper to pass a list of taints, I agree that it's best to avoid duplicating the removal code.

For generating the random taints, I went ahead and made a list for the tolerable taints on the first node + a map for each remaining node's intolerable taints. Let me know what you think